modules/appeals_api/app/workers/appeals_api/higher_level_review_pdf_submit_job.rb

```diff
@@ -13,11 +13,17 @@ class HigherLevelReviewPdfSubmitJob
 
     def perform(higher_level_review_id, retries = 0)
       higher_level_review = AppealsApi::HigherLevelReview.find(higher_level_review_id)
-      @retries = retries
-      stamped_pdf = AppealsApi::PdfConstruction::Generator.new(higher_level_review).generate
-      higher_level_review.update!(status: 'submitting')
-      upload_to_central_mail(higher_level_review, stamped_pdf)
-      File.delete(stamped_pdf) if File.exist?(stamped_pdf)
+
+      begin
+        @retries = retries
+        stamped_pdf = AppealsApi::PdfConstruction::Generator.new(higher_level_review).generate
+        higher_level_review.update!(status: 'submitting')
+        upload_to_central_mail(higher_level_review, stamped_pdf)
+        File.delete(stamped_pdf) if File.exist?(stamped_pdf)
+      rescue
+        higher_level_review.update!(status: 'error')
```

Comment:
Looking over this again, I feel like we should explore including a code, description, and source to this. Is there anything in `map_response` that might aid us?

Reply:
I'm not familiar with `map_response`. What are you referring to specifically?

Reply:
When we send the data to CentralMail we process its response and use `map_response` to figure out what `code`, `status`, and `description` to update the model with. I thought it might be a good jumping off point to supplying our own data into those same fields when non-CentralMail errors happen.

Even something simple like the `code` being "Exception" and the `description` having the error class & truncated message would be beneficial, I think.

Reply:
To be clear, the map_response looks for the error returned from `CentralMail`, and then populates the `AppealsApi::UploadError` which inherits from `CentralMail::UploadError`. It then populates the code, and status - necessary for initializing the UploadError. It also increments the StatsD error count for that specific error. 

I think we can entertain the idea of defining a custom catchall error for job failure, but at this point - it's not exactly clear what we're going to be rescuing... such as a syntax error, or serialization error as seen with the `perform_async(hlr)` issue we had earlier.

Reply:
Yeah, a catchall error is really all I was looking for, with the `description` populated with _something_ from the exception that was thrown so that we have at least some idea of what went on to narrow down our search within the logs.